### DISCUSSION TITLE
How to get value in bags or tables without GQL?

### DISCUSSION THREAD

**Discussion by krabat-h** [FrankC01/pysui#190] [Q&A] [answered]
how to print detail value in balances value?

    struct Assets has key, store {
        id: UID,
        balances: Bag,
    }
    
    the Bag has Balance<CoinType> values
    
    
   I saw function GetDynamicFieldObject, but I don't know how to construct the params. And GetDynamicFields return nothing.
   
![image](https://github.com/FrankC01/pysui/assets/121561767/baa9de40-1751-4a18-90b8-5efc7379ad36)

use `sui client dynamic-field` is ok:
![image](https://github.com/FrankC01/pysui/assets/121561767/084a6706-0c48-406f-a3b5-b1d4c0c8b3f7)

**Comment by FrankC01** [comment]
Change the `target_obj` to an object id of your choice
```python
def get_dynamic_fields():
    """Get dynamic field objects."""
    client = SyncClient(SuiConfig.default_config())
    target_obj = "0x6428faf5303575518c656326f138297d71b1982a2e37fad014fb1f851b3d50ea"
    result = client.execute(GetDynamicFields(target_obj))
    if result.is_ok():
        for dfield in result.result_data.data:
            if dfield.field_type == "DynamicObject":
                fmap = SuiMap("foo", "bar")
                fmap.map = dfield.name
                builder = GetDynamicFieldObject(target_obj, name=fmap)
                result = client.execute(builder)
                if result.is_ok():
                    print("Dynamic Object Field")
                    print(result.result_data.to_json(indent=2))
                else:
                    print(f"fail:{result.result_string}")
            else:
                print("Dynamic Field")
                print(dfield.to_json(indent=2))
    else:
        print(result.result_string)
```

**Reply by krabat-h** [reply]
got it, thanks a lot!

**Comment by krabat-h** [comment]
I have another question, could you please help me?

If I only have one gas coin object, how can I make a transaction about gas coin? As I know, gas coin object can not be in use, and split coin also need "not in use" object, but I only have 1 object.

**Comment by FrankC01** [comment]
You can use the `txb.gas` in that scenario

```python
    txb = SyncTransaction(client=client)
    # Split some portion from gas coin
    scres = txer.split_coin(coin=txb.gas, amounts=[30000000])
    # Use scres  in movecall or transfer back to yourself
    txb.transfer_objects(transfers=scres, recipient=client.config.active_address)
```

**Reply by krabat-h** [reply, marked as answer]
Thanks a lot, it works. So If I need to call a move func which need sui coin, I need to split sui coin to at least 2 object?
